tools/replace_code_block.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from traceback import print_tb
+from typing import AnyStr
+
+
+def check_for_code(filename: AnyStr, check: bool, print_not_fixed: bool):
+    """Check a file and replace ``.. code::`` to ``.. code-block::``
+
+    :param filename: The file to open
+    :param check: Whether or not the program is operating in ``check`` mode.
+        In check mode, the program will exit with a status 1 if files will be
+        replaced. In non-check mode, the program will replace files.
+    :param print_not_fixed: Print files that were not re-formatted, similar to
+        black.
+    """
+    status = 0
+    with open(filename, "rb") as fp:
+        old_content = fp.read()
+    new_content = old_content.replace(b".. code::", b".. code-block::")
+    if old_content == new_content:
+        if print_not_fixed:
+            print(
+                "File {filename} does not contain any ``code::`` "
+                "statements".format(filename=filename)
+            )
+        return 0
+    if check:
+        status = 1
+        print(
+            "File {filename} contains ``code::`` statements!".format(
+                filename=filename
+            )
+        )
+    else:
+        with open(filename, "wb") as fp:
+            fp.write(new_content)
+        print(
+            "Replaced ``code::`` statements to ``code-block::`` in "
+            "{filename}".format(filename=filename)
+        )
+    return status
+
+
+def _main(check, verbose):
```

Comment:
This method name isn't very descriptive for what is happening here. I think it's fine for this logic to be in `main`, but if it's going to be broken out, it'd be a good idea to have a useful name for it.

Reply:
in that case, I should copy-paste the code over